src/box/memtx_engine.cc

```diff
@@ -1341,36 +1341,18 @@ send_join_meta(struct xstream *stream, const struct raft_request *raft_req,
 #include "memtx_checkpoint_join.cc"
 #else /* !defined(ENABLE_FETCH_SNAPSHOT_CURSOR) */
 
-static int
-memtx_engine_prepare_checkpoint_join(struct engine *engine,
```

Comment:
Let's also drop `data` from the `engine_join_ctx` and allocation of that memory (in `engine_prepare_join`) . AFAIR, it was used only in `memtx_prepare_checkpoint_join`

https://github.com/tarantool/tarantool/blob/9a17476508ce0eb2562dfff1357281e25e274e15/src/box/engine.h#L361-L362

Reply:
Are you talking about [this commit](https://github.com/tarantool/tarantool/commit/72cc2b3ef7f2755ca7d611ee2f392f3311aad8a3#diff-42bfb43d7d5ce64ca78190e4c475ac8d6e17eddb90a44a43f7ea5cfe6e0f7754)? The member is only renamed there.

By the way, do we still need the above-mentioned refactoring, or should I revert it?

Reply:
No, I meant the same [commit](https://github.com/tarantool/tarantool/commit/72cc2b3ef7f2755ca7d611ee2f392f3311aad8a3#diff-42bfb43d7d5ce64ca78190e4c475ac8d6e17eddb90a44a43f7ea5cfe6e0f7754) in which you renamed the member. It was preparatory for your checkpoint join PR, so I'm making sure if it's still needed after I moved checkpoint-referencing part to CE.

Reply:
Probably, we should not do that. We decided to move part of `relay_initial_join` to the `memtx_engine_join`, as it was ugly inside the relay. We'll have to export new functions from EE in order to send checkpoint `JOIN_META` from relay, which majorly differs from the `JOIN_META` from read view and we didn't like that. Now it's ok, and I'd not go any further